src/MongoDB.EntityFrameworkCore/Serializers/SerializationHelper.cs

```diff
@@ -88,17 +90,30 @@ private static IBsonSerializer CreateTypeSerializer(IReadOnlyProperty property)
 
         var typeSerializer = CreateTypeSerializer(property.ClrType, property);
 
-        // Apply HasBsonType configuration if set
-        var bsonType = property.GetBsonType();
-        if (bsonType != null && typeSerializer is IRepresentationConfigurable representationConfigurable)
+        return property.GetBsonRepresentation() is { } bsonRepresentation
```

Comment:
Personally I do not like this weird way to check for not null value. IMHO, the following code doing the same but easier to read:
`var bsonRepresentation = property.GetBsonRepresentation();`
`return bsonRepresentation == null ? typeSerializer : ApplyBsonRepresentation(bsonRepresentation, typeSerializer);`

Reply:
I think I prefer the pattern matching version. Yes, in isolation it's a bit unusual but once you start using pattern matching everywhere it fits well with that.